packages/algorand-js/src/runtime/runtime.ts

```diff
@@ -184,6 +185,22 @@ export class Runtime {
     }
   }
 
+  // creates new application and returns application id
+  createApp (params: SSCDeploymentFlags): number {
+    const sender = params.sender;
+    const senderAccountDef = this.assertAccountDefined(this.store.accounts.get(sender.addr));
+    const app = senderAccountDef.createApp(++this.counter, params);
+
+    this.store.globalApps.set(app.id, app.params); // update globalApps Map
+    return app.id;
+  }
+
+  optInToApp (appId: number, accountAddr: string): void {
+    const appParams = this.store.globalApps.get(appId);
```

Comment:
you need to check if the app exists, and fail / throw an exception if it doesn't.

Reply:
added